service/provider/src/main/java/gov/va/vro/service/provider/MasPollingProcessor.java

```diff
@@ -16,27 +21,56 @@ public class MasPollingProcessor implements Processor {
 
   private final CamelEntrance camelEntrance;
   private final MasDelays masDelays;
+  private final MasApiService masService;
 
   @Override
   public void process(Exchange exchange) {
     var payload = exchange.getMessage().getBody(MasClaimDetailsPayload.class);
     log.info("Checking collection status for collection {}.", payload.getCollectionId());
     // call pcCheckCollectionStatus
-    boolean ready = checkCollectionStatus(payload.getCollectionId());
-    if (ready) {
-      log.info("Collection {} is ready for processing.", payload.getCollectionId());
-      // call pcQueryCollectionAnnots
-      // call Lighthouse
-      // Combine results and call PDF generation
-      // if a decision is made, call pcOrderExam
-    } else {
-      log.info("Collection {} is not ready. Requeueing...", payload.getCollectionId());
-      // re-request after some time
-      camelEntrance.notifyAutomatedClaim(payload, masDelays.getMasProcessingSubsequentDelay());
+    try {
+      boolean ready = checkCollectionStatus(payload.getCollectionId());
+      if (ready) {
+        log.info("Collection {} is ready for processing.", payload.getCollectionId());
+        // TODO:
+        // 1. call pcQueryCollectionAnnots
+        // 2. call Lighthouse
+        // 3. evaluate pcQueryCollectionAnnots and lighthouse result
+        //    boolean isEvidenceFound = evaluateMedicalEvidence(MasCollectionAnnotations,
+        // LightHousePatientMedicalInfo);
+        //    if (!isEvidenceFound) {
+        //         call masService.orderExam(payload.getCollectionId());
+        //         update queue to wait for order status.
+        //    } else { do step 4.}
+        // 4. Combine results and call PDF generation
+      } else {
+        log.info("Collection {} is not ready. Requeueing...", payload.getCollectionId());
+        // re-request after some time
+        camelEntrance.notifyAutomatedClaim(payload, masDelays.getMasProcessingSubsequentDelay());
+      }
+    } catch (MasException e) {
+      log.error("MAS collection {} status check failed.", payload.getCollectionId(), e);
     }
   }
 
-  private boolean checkCollectionStatus(String collectionsId) {
-    return new Random().nextBoolean();
+  private boolean checkCollectionStatus(String collectionsId) throws MasException {
+    // return new Random().nextBoolean();
+    try {
+      int masCollectionId = Integer.parseInt(collectionsId);
```

Comment:
I'll leave it for now, since there might be a whole lot of issues with the id when we can connect to MAS API and work with Lighthouse Fhir API. We can visit this later.